sandbox-backend/src/main/java/com/exadel/sandbox/team5/service/impl/SnsServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.exadel.sandbox.team5.service.impl;
+
+import com.amazonaws.services.sns.AmazonSNS;
+import com.amazonaws.services.sns.model.PublishResult;
+import com.exadel.sandbox.team5.dto.DiscountDto;
+import com.exadel.sandbox.team5.service.SnsService;
+import com.exadel.sandbox.team5.util.Message;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class SnsServiceImpl implements SnsService {
```

Comment:
Only it is better to use abstract name independent on realization. For example `NotificationService, NotificationServiceImpl`.

Reply:
done